sql/glean_clients_daily_v1.sql

```diff
@@ -0,0 +1,117 @@
+CREATE TEMP FUNCTION
+  udf_geo_struct(country STRING,
+    city STRING,
+    geo_subdivision1 STRING,
+    geo_subdivision2 STRING) AS ( IF(country IS NULL
+      OR country = '??',
+      NULL,
+      STRUCT(country,
+        NULLIF(city,
+          '??') AS city,
+        NULLIF(geo_subdivision1,
+          '??') AS geo_subdivision1,
+        NULLIF(geo_subdivision2,
+          '??') AS geo_subdivision2)));
+CREATE TEMP FUNCTION
+  udf_json_mode_last(list ANY TYPE) AS ((
+    SELECT
+      ANY_VALUE(_value)
+    FROM
+      UNNEST(list) AS _value
+    WITH
+    OFFSET
+      AS _offset
+    GROUP BY
+      TO_JSON_STRING(_value)
+    ORDER BY
+      COUNT(_value) DESC,
+      MAX(_offset) DESC
+    LIMIT
+      1));
+CREATE TEMP FUNCTION
+  udf_mode_last(list ANY TYPE) AS ((
+    SELECT
+      _value
+    FROM
+      UNNEST(list) AS _value
+    WITH
+    OFFSET
+      AS
+    _offset
+    GROUP BY
+      _value
+    ORDER BY
+      COUNT(_value) DESC,
+      MAX(_offset) DESC
+    LIMIT
+      1 ));
+--
+WITH
+  base AS (
+  SELECT
+    DATE(submission_timestamp) AS submission_date,
+    LOWER(client_info.client_id) AS client_id,
+    *
+  FROM
+    `moz-fx-data-shared-prod.org_mozilla_fenix.metrics_v1`
+  WHERE
+    client_info.client_id IS NOT NULL ),
+  --
+  numbered_duplicates AS (
+  SELECT
+    *,
+    ROW_NUMBER() OVER (PARTITION BY client_id, submission_date, document_id ORDER BY submission_timestamp) AS _n
+  FROM
+    base ),
+  -- Deduplicating on document_id is necessary to get valid SUM values.
+  deduplicated AS (
+  SELECT
+    * EXCEPT (_n)
+  FROM
+    numbered_duplicates
+  WHERE
+    _n = 1 ),
+  windowed AS (
+  SELECT
+    submission_date,
+    client_id,
+    ROW_NUMBER() OVER w1_unframed AS _n,
+    MIN(SAFE.PARSE_DATE('%F', SUBSTR(client_info.first_run_date, 1, 10))) OVER w1 AS profile_date,
+    -- For all other dimensions, we use the mode of observed values in the day.
+    'Fenix' AS app_name,
+    udf_mode_last(ARRAY_AGG(client_info.os) OVER w1) AS os,
+    udf_json_mode_last(ARRAY_AGG(udf_geo_struct(metadata.geo.country, metadata.geo.city, NULL, NULL)) OVER w1).* EXCEPT (geo_subdivision1, geo_subdivision2),
+    udf_mode_last(ARRAY_AGG(client_info.app_build) OVER w1) AS app_build_id,
+    IF(udf_mode_last(ARRAY_AGG(client_info.app_build) OVER w1) IN ('11721515', '11721514'), 'release', 'nightly') AS channel,
```

Comment:
Do we have documentation about the build IDs and why 11721515 and 11721514 are considered release?

Reply:
None that I know of, but this is probably a bad assumption to make when we create another release.

Reply:
It looks like `client_info.app_channel` is nearly always "release", with a few NULLs. Do you have any other ideas of how to reliably determine channel?

Reply:
Perhaps the nulls all correspond to nightly?

Reply:
It looks like nulls are for build starting in 113 and below. All newer builds show up with app_channel = release.

Reply:
We discussed elsewhere, and it looks like fenix nightly is going to get its own app name and thus separate tables. We'll union those together in a future update.